lock-api/src/main/java/com/palantir/lock/TimelockNamespace.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface TimelockNamespace {
+    String value();
```

Comment:
In #4042 you should add `@JsonSerialize/@JsonDeserialize` and `@JsonValue` so you don't have to have extra methods to convert `Set<String>` to `Set<TimelockNamespace>`.

Reply:
Will be adding it to following pr.